models/tsodyks2_synapse.h

```diff
@@ -227,9 +231,19 @@ tsodyks2_synapse< targetidentifierT >::send( Event& e, size_t t, const CommonSyn
 {
   Node* target = get_target( t );
   const double t_spike = e.get_stamp().get_ms();
-  const double h = t_spike - t_lastspike_;
-  double x_decay = std::exp( -h / tau_rec_ );
-  double u_decay = ( tau_fac_ < 1.0e-10 ) ? 0.0 : std::exp( -h / tau_fac_ );
+
+  if ( t_lastspike_ >= 0.0 )
+  {
+    // only update x and u if this is not the first spike to pass through the synapse
+
+    const double h = t_spike - t_lastspike_;
+    double x_decay = std::exp( -h / tau_rec_ );
+    double u_decay = ( tau_fac_ < 1.0e-10 ) ? 0.0 : std::exp( -h / tau_fac_ );
```

Comment:
```suggestion
    const double x_decay = std::exp( -h / tau_rec_ );
    const double u_decay = ( tau_fac_ == 0 ) ? 0.0 : std::exp( -h / tau_fac_ );  // tau_fac == 0 disables facilitation
```
I find this check here a bit confusing. Since `tau_fac_ == 0` is allowed to turn off facilitation (actually the default), I think it is most sensible to actually test for zero.

Reply:
There is the risk of a floating point underflow, but I don't think that will be a problem in practice (by default it shouldn't throw an exception).